### PR TITLE
Update ZIO to 1.0.0-RC18

### PR DESCRIPTION
**Pull request by ioleo**
Same as with [interop-cats#145](https://github.com/zio/interop-cats/pull/145). Been playing around and needed to update `zio-config` to latest ZIO. Tested only on `2.13.1`.

This PR is meant to hang around open until next ZIO release. I'll try to keep it up to date as ZIO Core evolves.

### CODE REVIEW

File: core/src/main/scala/zio/config/ConfigDescriptor.scala [hunk 1]
```diff
@@ -6,60 +6,45 @@ import java.net.URI
 import zio.config.ConfigDescriptor._
```
Comment (ioleo) [inline core/src/main/scala/zio/config/ConfigDescriptor.scala]: All I did here is to sort methods alphabetically.

File: core/src/main/scala/zio/config/package.scala [hunk 2]
```diff
@@ -1,11 +1,14 @@
 package zio
 
 package object config extends ReadFunctions with WriteFunctions with ConfigDocsFunctions {
+
+  type Config[A] = Has[Config.Service[A]]
```
Comment (ioleo) [inline core/src/main/scala/zio/config/package.scala]: Following the convention from ZIO Core.

Comment (afsalthaj) [reply core/src/main/scala/zio/config/package.scala]: Sure 👍

**Comment by afsalthaj** [comment]
Thanks @ioleo for the PR. Looking forward

**Comment by ioleo** [comment]
Starting to upgrade the PR to `1.0.0-RC18`.

**Pull request merged**